### PR TITLE
Docs/dont load sklearn detectors in notebooks

### LINKED ISSUE #827: The kddcup isolation forest artefact needs updating with `sklearn 1.3.0`
The `fetch_detector` call run by `od_if_kddcup.ipynb` returns the following warning (which leads to an error):

```
 InconsistentVersionWarning: Trying to unpickle estimator ExtraTreeRegressor from version 1.0 when using version 1.3.0. This might lead to breaking code or invalid results. Use at your own risk.
```

It is likely we need to regenerate `https://console.cloud.google.com/storage/browser/seldon-models/alibi-detect/od/IForest/kddcup` with `sklearn 1.3.0`.

### PR DESCRIPTION
**Pull request by jklaise**
Resolves #827 by completely removing calls to old, pre-trained `sklearn` detectors. The training of the detectors from scratch only takes a few seconds at most.

### CODE REVIEW

**Comment by ascillitoe** [comment]
Why didn't I think of this! Much easier than constantly maintaining those artefacts...

File: doc/source/examples/od_if_kddcup.ipynb
[the remark this answers is not in the document]
Comment (jklaise) [reply doc/source/examples/od_if_kddcup.ipynb:1]: Ah damn classic mistake, will fix. We should have an automated way of removing this but would tricky to set up.

[the remark this answers is not in the document]
Comment (ascillitoe) [reply doc/source/examples/od_if_kddcup.ipynb:1]: Yeh agreed. Didn't @mauicv look into [nbstripout](https://github.com/kynan/nbstripout) or similar a while back?

[the remark this answers is not in the document]
Comment (mauicv) [reply doc/source/examples/od_if_kddcup.ipynb:1]: Yeah (see [here](https://github.com/SeldonIO/alibi/pull/644)), as I recall it did %90 of what we wanted but not everything.

**Pull request merged**